### ISSUE TITLE
Nullness Checker false negative when instantiating inner classes with enclosing expression

### ISSUE THREAD

**Issue by wmdietl** [eisop/checker-framework#281]
Take this simple code:

````
import org.checkerframework.checker.nullness.qual.Nullable;

class InnerCrash {
    class Inner {}

    static @Nullable InnerCrash getInnerCrash() {
        return null;
    }

    static void foo() {
        Object o = getInnerCrash().new Inner();
    }

    public static void main(String[] args) {
        foo();
    }
}
````

The Nullness Checker doesn't issue a warning about this code, but executing it results in a NPE:

````
Exception in thread "main" java.lang.NullPointerException
	at java.base/java.util.Objects.requireNonNull(Objects.java:221)
	at InnerCrash.foo(InnerCrash.java:11)
	at InnerCrash.main(InnerCrash.java:15)
````

https://github.com/eisop/checker-framework/blob/d6b7519290ccdc3579056d7d6f0070e00c4ac3eb/checker/src/main/java/org/checkerframework/checker/nullness/NullnessVisitor.java#L723
should ensure the enclosing expression is non-null.
